driver-core/src/main/com/mongodb/internal/operation/ListSearchIndexesOperation.java

```diff
@@ -0,0 +1,134 @@
+import static com.mongodb.internal.operation.CommandOperationHelper.rethrowIfNotNamespaceError;
+import static com.mongodb.internal.operation.OperationHelper.createEmptyAsyncBatchCursor;
+import static com.mongodb.internal.operation.OperationHelper.createEmptyBatchCursor;
+
+/**
+ * An operation that lists Alas Search indexes with the help of {@value #STAGE_LIST_SEARCH_INDEXES} pipeline stage.
+ *
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public class ListSearchIndexesOperation<T>
+        implements AsyncExplainableReadOperation<AsyncBatchCursor<T>>, ExplainableReadOperation<BatchCursor<T>> {
+    private static final String STAGE_LIST_SEARCH_INDEXES = "$listSearchIndexes";
+    private final MongoNamespace namespace;
+    private final Decoder<T> decoder;
+    private final Boolean allowDiskUse;
+    private final Integer batchSize;
+    private final Collation collation;
+    private final BsonValue comment;
+    private final long maxAwaitTimeMS;
+    private final long maxTimeMS;
+    private final String indexName;
+
+    public ListSearchIndexesOperation(final MongoNamespace namespace,
+                                      final Decoder<T> decoder,
+                                      final long maxTimeMS, final long maxAwaitTimeMS,
+                                      @Nullable final String indexName,
+                                      @Nullable final Integer batchSize,
+                                      @Nullable final Collation collation,
+                                      @Nullable final BsonValue comment,
+                                      @Nullable final Boolean allowDiskUse) {
+        this.namespace = namespace;
+        this.decoder = decoder;
+        this.maxTimeMS = maxTimeMS;
+        this.maxAwaitTimeMS = maxAwaitTimeMS;
+        this.batchSize = batchSize == null ? Integer.valueOf(0) : batchSize;
+        this.collation = collation;
+        this.comment = comment;
+        this.allowDiskUse = allowDiskUse;
+        this.indexName = indexName;
+    }
+
+    @Override
+    public BatchCursor<T> execute(final ReadBinding binding) {
+        try {
+            return getAggregateOperation().execute(binding);
+        } catch (MongoCommandException exception) {
+            return assertNotNull(rethrowIfNotNamespaceError(exception, createEmptyBatchCursor(namespace, decoder,
+                    exception.getServerAddress(), batchSize)));
+        }
+    }
+
+    @Override
+    public void executeAsync(final AsyncReadBinding binding, final SingleResultCallback<AsyncBatchCursor<T>> callback) {
+        getAggregateOperation().executeAsync(binding, (cursor, exception) -> {
+            if (exception != null && !isNamespaceError(exception)) {
+                callback.onResult(null, exception);
+            } else if (exception == null) {
+                callback.onResult(cursor, null);
```

Comment:
In this specific case, the variable `cursor` is guaranteed to never be `null.` This assurance comes from the fact that the implementation of [CommandReadTransformerAsync](https://github.com/mongodb/mongo-java-driver/blob/bb610dd5b33f2057c6af0ab949705f92478f51ac/driver-core/src/main/com/mongodb/internal/operation/AggregateOperationImpl.java#L252) always returns a result.

To provide further context, the transformation process is implemented in [CommandOperationHelper.java#L338](https://github.com/mongodb/mongo-java-driver/blob/123c8cdc8a618ce6306d7afd090fcfe76e88afbc/driver-core/src/main/com/mongodb/internal/operation/CommandOperationHelper.java#L338).